src/hotspot/share/jfr/support/jfrThreadLocal.cpp

```diff
@@ -537,3 +541,48 @@ Arena* JfrThreadLocal::dcmd_arena(JavaThread* jt) {
   tl->_dcmd_arena = arena;
   return arena;
 }
+
+
+#ifdef LINUX
+
+bool JfrThreadLocal::acquire_cpu_time_jfr_enqueue_lock() {
```

Comment:
Can you describe the lock protocol and why a thread must lock its thread-local queue?

Is it because sampling interrupts can interleave after you have started to process a sample as part of a previous interrupt?

Reply:
Yes and they can interleave with the safepoint too.